airflow/providers/microsoft/azure/provider.yaml

```diff
@@ -77,11 +77,9 @@ dependencies:
   - azure-synapse-spark
   - adal>=1.2.7
   - azure-storage-file-datalake>=12.9.1
-  # TODO: upgrade to newer versions of all the below libraries.
-  #   See issue https://github.com/apache/airflow/issues/30199
-  - azure-mgmt-containerinstance>=1.5.0,<2.0
-  - azure-mgmt-datafactory>=1.0.0,<2.0
-  - azure-kusto-data>=0.0.43,<0.1
+  - azure-mgmt-containerinstance>=2.0
+  - azure-mgmt-datafactory>=2.0
+  - azure-kusto-data>=0.1
```

Comment:
I don't think it would be easy to address all of these in 1 PR.
See initial attempt 
https://github.com/apache/airflow/pull/29764

Reply:
You are right, I knew it wasn't going to be so smooth. 
But I am willing to fix the issues and bump up the Azure SDK to make it work with latest APIs.

Reply:
Yeah. Let's iterate on it until it's green :)